compose-v2/galaxy-postgres/Dockerfile

```diff
@@ -0,0 +1,32 @@
+FROM library/postgres:11.2 AS build
+
+ARG GALAXY_RELEASE=release_19.09
+ARG GALAXY_REPO=https://github.com/galaxyproject/galaxy
+
+COPY init-galaxy-db.sql.in /docker-entrypoint-initdb.d/
+COPY init-galaxy-db.sh /docker-entrypoint-initdb.d/
+RUN chmod a+rx /docker-entrypoint-initdb.d/init-galaxy-db.sh && \
+    chmod a+r /docker-entrypoint-initdb.d/init-galaxy-db.sql.in
+
+# Initialize database with Galaxy model to create initial sql-dump
+ENV POSTGRES_PASSWORD=build \
+    POSTGRES_USER=galaxy \
+    POSTGRES_DB=galaxy
+# Install Galaxy (there needs to be a better way!!)
+RUN apt update && apt install --no-install-recommends build-essential ca-certificates curl gcc libpq-dev python-pip python-dev -y \
+    && rm -rf /var/lib/apt/lists/* \
+    && mkdir /galaxy && curl -L -s $GALAXY_REPO/archive/$GALAXY_RELEASE.tar.gz | tar xzf - --strip-components=1 -C /galaxy \
+    && cd /galaxy && ./scripts/common_startup.sh --skip-client-build && cd / \
+    && . /galaxy/.venv/bin/activate && pip install psycopg2 && deactivate
+COPY files/galaxy.yml /galaxy/config/galaxy.yml
+RUN docker-entrypoint.sh postgres & sleep 10 \
+    && /galaxy/create_db.sh -c /galaxy/config/galaxy.yml \
+    && PGPASSWORD=$POSTGRES_PASSWORD pg_dump -h localhost -U galaxy galaxy > /docker-entrypoint-initdb.d/galaxy.sql
+
+FROM library/postgres:11.2
+
+COPY init-galaxy-db.sql.in /docker-entrypoint-initdb.d/
```

Comment:
It's not necessary to include database dumps, since 19.01 Galaxy will just create what's needed if given an empty database.

Reply:
You are right that it is not really necessary (in that it also works without it), but in my experiments this speeds up the first start of Galaxy quite a bit.

Reply:
It should be around 200ms, and we do this hundreds of times during Galaxy's integration tests. Can you check this again and report what you're seeing?

Reply:
Strange.. In my set up it takes around 3 minutes (see: https://github.com/AndreasSko/docker-galaxy-stable/runs/436011222). Is the integration test for Galaxy using SQLite or Postgres?

Reply:
This run goes through the migrations because the database exists. if you let Galaxy create the table there will be no migrations. I'll have a look at doing that for empty databases as well